File: fix_session.ts
import { db } from "./src/server/firebase_client";
import { doc, updateDoc, setDoc } from "firebase/firestore";

async function run() {
  await setDoc(doc(db, "bot_sessions", "551085277"), {
    chatId: 551085277,
    userId: "havpOEUkVkSXtSWZbJWHEQpsqQH2",
    email: "gowtham.jaihind@gmail.com",
    linkedAt: Date.now(),
    lastSeenAt: Date.now()
  }, { merge: true });

  await updateDoc(doc(db, "users", "havpOEUkVkSXtSWZbJWHEQpsqQH2"), {
    telegramChatId: 551085277,
    telegramLinkedAt: Date.now()
  });

  console.log("Fixed session & user");
  process.exit(0);
}
run().catch(console.error);

File: fix_session.py
import os

with open("src/server/telegram/session.ts", "r") as f:
    text = f.read()

text = text.replace("import { store } from \"./session\";\n  const sessions = (store as any).sessions || {};\n  delete sessions[chatId];", "await setDoc(doc(db, \"bot_sessions\", String(chatId)), { userId: deleteField(), email: deleteField(), orgId: deleteField() }, { merge: true });")

with open("src/server/telegram/session.ts", "w") as f:
    f.write(text)

